### PR TITLE
Timed macro

### PR DESCRIPTION
**Pull request by Quaqqer**
Add a macro for timing functions.

Future improvements
- Printing takes time and will cause nested timed functions to execute slower => store in structure and print afterwards
- More information about execution time

Try with
```bash
$ RUSTFLAGS="--cfg timed" cargo run <file>
```

### CODE REVIEW

File: sylt-macro/src/macro.rs [hunk 1]
```diff
@@ -131,3 +131,53 @@ pub fn derive_numbered(item: proc_macro::TokenStream) -> proc_macro::TokenStream
...
+/// Timed macro
+///
+/// Time a function and print the output.
+///
+/// # Examples
+///
+/// ```
+/// #[sylt_macro::timed]
+/// fn hej() {
+///     println!("Hej");
```
Comment (FredTheDino) [inline sylt-macro/src/macro.rs:144]: Maybe not in swedish?
```suggestion
/// fn hi() {
///     println!("Hi");
```

File: sylt-macro/src/macro.rs [hunk 2]
```diff
@@ -131,3 +131,53 @@ pub fn derive_numbered(item: proc_macro::TokenStream) -> proc_macro::TokenStream
...
+/// }
+/// ```
+/// will output `Time::hej = 123μ`
```
Comment (FredTheDino) [inline sylt-macro/src/macro.rs:147]:
```suggestion
/// will output `Time::hi = 123μ`
```

File: sylt-macro/src/macro.rs [hunk 3]
```diff
@@ -131,3 +131,53 @@ pub fn derive_numbered(item: proc_macro::TokenStream) -> proc_macro::TokenStream
...
+
+    let new_function_block = quote! {
+        {
+            let __timer_start = std::time::SystemTime::now();
+
+            let v = (move || {
+                #function_block
+            })();
+
+            println!("Time::{} = {:?}", #signature, __timer_start.elapsed().unwrap());
```
Comment (FredTheDino) [inline sylt-macro/src/macro.rs:174]:
```suggestion
            eprintln!("Time::{} = {:?}", #signature, __timer_start.elapsed().unwrap());
```
Print it on the error-stream? Does this write out a mu?

**Review by FredTheDino — APPROVED** [review]
LGTM, make it english and it looks great!

**Comment by sornas** [comment]
will want to take a look later tonight

**Review by sornas — APPROVED** [review]
i thought something looked off but i was wrong. looks great!

**Pull request merged**